sdk/storage_datalake/src/operations/path_head.rs

```
@@ -87,18 +87,18 @@ pub struct HeadPathResponse {
     pub common_storage_response_headers: CommonStorageResponseHeaders,
     pub etag: String,
     pub last_modified: DateTime<Utc>,
-    pub properties: Properties,
+    pub properties: Option<Properties>,
 }
 
 impl HeadPathResponse {
     pub async fn try_from(response: HttpResponse) -> Result<Self, crate::Error> {
-        let (_status_code, headers, _pinned_stream) = response.deconstruct();
+        let headers = response.headers();
 
         Ok(Self {
-            common_storage_response_headers: (&headers).try_into()?,
-            etag: etag_from_headers(&headers)?,
-            last_modified: last_modified_from_headers(&headers)?,
-            properties: (&headers).try_into()?,
+            common_storage_response_headers: headers.try_into()?,
+            etag: etag_from_headers(headers)?,
+            last_modified: last_modified_from_headers(headers)?,
+            properties: headers.try_into().ok(),
```

Thanks for the details on why it is Option<Properties> now, but why use `.ok()` here instead of `?`?

---

Ahh, that's because I need to convert `Result<T, E>` to `Option<T>` since `properties` is now an `Option`.  However, I just realized a problem with this whereby the `E` get's discarded silently in this implementation. A better implementation would be to only discard the error if it is `HeaderNotFound` since this is now an expected condition, and propagate all other errors that may come out of `try_into()`.  So, basically this:

```rust
properties: match headers.try_into() {
    Ok(p) => Some(p),
    // 'x-ms-properties' will not be returned in some cases, such as when the request 'action' is set to 'getStatus'
    Err(crate::Error::HeaderNotFound(_)) => None,
    // Propogate all other errors
    Err(e) => return Err(e),
},
```

I just tested this locally and it works as expected.  If you're OK with this implementation, I'll push it into my PR.

---

@cataggar, just heads up that I went ahead and applied the change above. Let me know if this resolves your concerns.

---

Sorry @rickrain, I'm lacking time to dig into this. An error case shouldn't be a normal workflow. If the header is optional for an operation and it is not found, it should return None. It should return an error and then map that to None.

---

Ok.  So, you would rather me write the same logic that `try_into()` already does so I can check if the header is there first, rather than letting `try_into()` just return the error and respond accordingly. This seems to be very opinionated, but ok.  In other languages, where _exceptions_ are expensive, I would agree with this opinion. But, I'm not sure I'm in agreement here given how errors are handled in Rust.

---

You can use get_option_str_from_headers https://github.com/rickrain/azure-sdk-for-rust/pull/3 .

---

I had no idea that function existed. Thank you for steering me in the right direction!  Had a different implementation I was about to commit but like yours better given it's using an existing function from the SDK.